src/operator/subgraph/subgraph_op.cc

```diff
@@ -70,16 +91,88 @@ struct SubgraphOpState {
   std::vector<std::string> mutable_data_names_;
   //std::vector<std::string> input_data_names_;
   std::vector<std::string> output_data_names_;
-  std::string subgraph_uuid_;
   std::shared_ptr<Executor> subgraph_executor_;
-};  // SubgraphOpState
+};
+
+SubgraphOperatorPtr SimpleSubgraphProperty::CreateSubgraphOperator(const nnvm::Symbol &sym) const {
+  return std::make_shared<DefaultSubgraphOperator>(sym);
+}
+
+void DefaultSubgraphOperator::Forward(const OpContext& ctx,
+                                      const std::vector<NDArray>& inputs,
+                                      const std::vector<OpReqType>& req,
+                                      const std::vector<NDArray>& outputs) {
+  // We can create an executor to run this subgraph op
```

Comment:
I will change the implementation here to adopt CachedOp since we want to support varied batch size seamlessly.